quic/s2n-quic-transport/src/recovery/manager.rs

```diff
@@ -333,12 +333,27 @@ impl<Config: endpoint::Config> Manager<Config> {
         &mut self,
         timestamp: Timestamp,
         frame: frame::Ack<A>,
+        packet_number: PacketNumber,
         random_generator: &mut Config::RandomGenerator,
         context: &mut Ctx,
         publisher: &mut Pub,
     ) -> Result<(), transport::Error> {
         let space = self.space;
         let largest_acked_packet_number = space.new_packet_number(frame.largest_acknowledged());
+
+        let path_id = context.path_id();
+        let path = context.path_mut();
+        for ack_range in frame.ack_ranges() {
```

Comment:
This needs to happen in the `process_ack_range` function, otherwise we'll be decoding the frame twice